File: Task 1/Java/app/src/main/java/dev/ogabek/java/database/NoteRepository.java
package dev.ogabek.java.database;

import android.app.Application;

import java.util.List;

import dev.ogabek.java.manager.RoomManager;
import dev.ogabek.java.model.Note;

public class NoteRepository {

    private NoteDao noteDao;

    public NoteRepository(Application application) {
        RoomManager db = RoomManager.getDatabase(application);
        noteDao = db.noteDao();
    }

    public List<Note> getNotes() {
        return noteDao.getAllNotes();
    }

    public void saveNote(Note note) {
        noteDao.saveData(note);
    }

    public void clearData() {
        noteDao.clearData();
    }

}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/database/NoteRepository.kt
package dev.ogabek.kotlin.database

import android.app.Application
import dev.ogabek.kotlin.manager.RoomManager
import dev.ogabek.kotlin.model.Note

class NoteRepository(application: Application) {

    private val db = RoomManager.getDatabase(application)
    private val noteDao: NoteDao = db.noteDao()

    fun getNotes(): List<Note> {
        return noteDao.getAllNotes()
    }

    fun saveNote(note: Note) {
        noteDao.saveNote(note)
    }

    fun deleteData() {
        noteDao.deleteData()
    }

}